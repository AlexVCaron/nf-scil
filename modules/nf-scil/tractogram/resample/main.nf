process TRACTOGRAM_RESAMPLE {
    tag "$meta.id"
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://scil.usherbrooke.ca/containers/scilus_2.0.2.sif':
        'scilus/scilus:2.0.2' }"

    input:
        tuple val(meta), path(centroids)

    output:
        tuple val(meta), path("*_centroid_{*}.{trk,tck}")   , emit: centroids
        path "versions.yml"                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def nb_points = task.ext.nb_points ? "${task.ext.nb_points}" : "5"

    """
    for centroid in ${centroids};
        do
        ext=\${centroid#*.}

        if [[ \$centroid == *"__"* ]]; then
            pos=\$((\$(echo \$centroid | grep -b -o __ | cut -d: -f1)+2))
            bname=\${centroid:\$pos}
            bname=\$(basename \$bname .\${ext})
        else
            bname=\$(basename \$centroid .\${ext})
        fi
        bname=\${bname/_centroid/}
        bname=\${bname/_ic/}

        scil_tractogram_resample_nb_points.py \$centroid \
            "${prefix}__\${bname}_centroid_${nb_points}.\${ext}" \
            --nb_pts_per_streamline $nb_points -f
    done

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: 2.0.2
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    scil_tractogram_resample_nb_points.py -h

    for centroid in ${centroids};
        do \
        ext=\${centroid#*.}
        bname=\$(basename \${centroid} .\${ext})
        touch ${prefix}__\${bname}${suffix}.\${ext}
    done

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: 2.0.2
    END_VERSIONS
    """
}
