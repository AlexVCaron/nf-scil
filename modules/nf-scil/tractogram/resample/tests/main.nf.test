nextflow_process {

    name "Test Process TRACTOGRAM_RESAMPLE"
    script "../main.nf"
    process "TRACTOGRAM_RESAMPLE"

    tag "modules"
    tag "modules_nfcore"
    tag "tractogram"
    tag "tractogram/resample"

    tag "subworkflows/load_test_data"

    setup {
        run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
            script "../../../../../subworkflows/nf-scil/load_test_data/main.nf"
            process {
                """
                input[0] = Channel.from( [ "bundles.zip", "filtering.zip" ] )
                input[1] = "test.load-test-data"
                """
            }
        }
    }

    test("tractogram - resample") {

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_bam'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("sarscov2 - bam - stub") {

        options "-stub"

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_bam'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }

    }

}
