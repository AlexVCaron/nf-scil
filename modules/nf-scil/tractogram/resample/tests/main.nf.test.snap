{
    "tractogram - resample - centroids": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_IFGWM_uni_c_10_resampled.trk:md5,5a4c819b4aa7ea048a4b89973ede812c"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f593a3a81a06f02aaa1ba17ba0baccfd"
                ],
                "tractograms": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_IFGWM_uni_c_10_resampled.trk:md5,5a4c819b4aa7ea048a4b89973ede812c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f593a3a81a06f02aaa1ba17ba0baccfd"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-22T11:23:01.440528"
    },
    "tractogram - resample - tractogram": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_IFGWM_color_resampled.trk:md5,65c7b9fc087c2289d928196244e2a16c",
                            "test_IFGWM_resampled.trk:md5,65c7b9fc087c2289d928196244e2a16c"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,f593a3a81a06f02aaa1ba17ba0baccfd"
                ],
                "tractograms": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_IFGWM_color_resampled.trk:md5,65c7b9fc087c2289d928196244e2a16c",
                            "test_IFGWM_resampled.trk:md5,65c7b9fc087c2289d928196244e2a16c"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f593a3a81a06f02aaa1ba17ba0baccfd"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-22T11:05:08.020285"
    }
}